File: 2022-2023/sourcecode/lab/lab12/ModulesBasicsJava/app/src/main/java/pl/edu/uwr/pum/modulesbasicsjava/di/modules/ComputerModule.java
package pl.edu.uwr.pum.modulesbasicsjava.di.modules;

import dagger.Module;
import dagger.Provides;
import pl.edu.uwr.pum.modulesbasicsjava.computer.Computer;
import pl.edu.uwr.pum.modulesbasicsjava.computer.components.CPU;
import pl.edu.uwr.pum.modulesbasicsjava.computer.components.GPU;

@Module
public class ComputerModule {
    @Provides
    static GPU provideGpu(){
        return new GPU();
    }

    @Provides
    static CPU provideCpu(){
        return new CPU();
    }

    @Provides
    static Computer provideComputer(GPU gpu, CPU cpu){
        return new Computer(gpu, cpu);
    }
}

File: 2022-2023/sourcecode/lab/lab12/ModulesBasicsKotlin/app/src/main/java/pl/edu/uwr/pum/modulesbasicskotlin/di/modules/ComputerModule.kt
package pl.edu.uwr.pum.modulesbasicskotlin.di.modules

import dagger.Module
import dagger.Provides
import pl.edu.uwr.pum.modulesbasicskotlin.computer.Computer
import pl.edu.uwr.pum.modulesbasicskotlin.computer.components.CPU
import pl.edu.uwr.pum.modulesbasicskotlin.computer.components.GPU

@Module
class ComputerModule {

    @Provides
    fun provideGpu(): GPU{
        return GPU()
    }

    @Provides
    fun provideCpu(): CPU{
        return CPU()
    }

    @Provides
    fun provideComputer(cpu: CPU, gpu: GPU): Computer{
        return Computer(gpu, cpu)
    }
}